### PR TITLE
New REST API tests

### PR DESCRIPTION
**Pull request by mkoura**
* Adding tests for automate and providers collections and for basic REST API functionality
* Removing redundant test `test_edit_user_password`
* Removing support for old REST API versions and old CFME versions
* Fixing behavior when entity doesn't contain "href" and contains "id"

__PRT outcome explained:__
* 5.7:
__1 failed__: timeout ->`Provider vsphere55 is behaving badly, marking it as bad.`
__6 errors__: `Exception: No providers could be set up matching the params`

* upstream:
`SproutException: Exception DoesNotExist raised! AppliancePool matching query does not exist.`

i.e. no problems caused by PR, can be approved and merged ;)

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit c9f147f6e617e235168198bc7f69be777ca0662f:

`cfme/tests/infrastructure/test_providers.py`:
- :bangbang: Line 264:5: **E303** *too many blank lines (2)*
- :bangbang: Line 274:5: **E303** *too many blank lines (2)*
- :bangbang: Line 294:5: **E303** *too many blank lines (2)*

`cfme/tests/test_rest.py`:
- :red_circle: Line 3:1: **F401** *'fauxfactory' imported but unused*
- :red_circle: Line 7:1: **F401** *'Credential' imported but unused*
- :red_circle: Line 8:1: **F401** *'Group' imported but unused*
- :red_circle: Line 8:1: **F401** *'User' imported but unused*
- :red_circle: Line 9:1: **F401** *'login' imported but unused*
- :red_circle: Line 9:1: **F401** *'login_admin' imported but unused*
- :red_circle: Line 13:1: **F401** *'conf' imported but unused*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6ce89d147f735b8228e02c6f4220ba05a94dde9b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: utils/api.py [hunk 1]
```diff
@@ -237,29 +232,12 @@ def reload_if_needed(self):
...
-            except APIException:
-                return self._find_by_filter(**params)
-        elif self._api.version.is_in_series("1.1") or self._api.version >= "2.0.0":
-            # New function
-            return self._find_by_filter(**params)
-        else:
-            # Old function
-            return self._find_by_sqlfilter(**params)
-
-    def _find_by_sqlfilter(self, **params):
```
Comment (psav) [inline utils/api.py]: These are replaced? We should probably use the removed in there for a while esp if there is usage of it somewhere in the codebase

**Comment by dajoRH** [comment]
Lint report for commit dd32b35fc5edec9be9fb2ac298bbcf0d66476e64:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f4c2c77b55ddf6226037d229deb6cebcee16b09b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: utils/api.py (hunk 1 above)
Comment (mkoura) [inline utils/api.py]: `_find_by_sqlfilter` is no longer needed, it was valid for old versions of api that is no longer used by any version of CFME that we test. Also the logic for choosing appropriate method can be removed and what used to be `_find_by_filter` internal method can now be directly `find_by`. `_find_by_sqlfilter` and `_find_by_filter` internal methods are not used directly anywhere, always through `find_by`, so it's safe to remove them.

Comment (mkoura) [inline utils/api.py]: @psav are there more clarifications and/or changes needed?

**Comment by mfalesni** [comment]
@mkoura you should now pick up that my PR which removes the builtin api client and uses the externalized one.

**Pull request merged**